Random cleanups

fskreuz opened the pull request:
## Description of the pull request:

Random cleanups:

- Removing the `clean` script since I discovered that the same can be done using [`git clean -fdx`](http://stackoverflow.com/a/64966/575527).
- Removing `./node_modules/.bin` from the scripts. [npm adds `node_modules/.bin` to `PATH`](https://docs.npmjs.com/cli/run-script) anyways.
- Removes [Testling ](https://ci.testling.com/) from package.json since Testling appears to be non-functional.
- Removes [Jam](https://github.com/caolan/jam) from package.json since it appears to be inactive.
- Removes the `component.json` in `scripts/templates`
- Removes the fluff in the root package.json since it's only used for build and not the one published.
- Minor reorganizing of package.json while I'm in there.

## Fixes the following issues:

An intimidating codebase 😄 

## Is breaking:

- If some build process used the newly added `clean`.
- If anyone is still using Jam as their package manager.
- If anyone is still using Component as their package manager.
- If there's something in `.gitignore` that should have been ignored/not ignored.
- If there's a reason why `./node_modules/.bin` was in the paths in the first place.
- If anyone is using `npm install ractivejs/ractive` (pulls repo as a dependency)
    - Suggesting you use `npm install ractivejs/ractive#TAG` to pull in the built versions.

## Reviewers:

Everyone, in particular those who know what goes on in the CI realm.

evs-chris commented:
I recently discovered that the package manifest in the root dir isn't the one that gets published to npm. The manifest that gets published is a template in the scripts dir that gets adjusted during the build process.

LGTM :+1:

Commit:
Reverse order of message and headings in Github helper files

Commit:
Update package manager manifests to be consistent with the root package.json

Commit:
Update ignored files and directories

Commit:
Update package.json. Cleanups and reorganization.

Commit:
Trim down root package.json down to building only

Following the lead of Lodash. Since the package.json that gets published
isn't the one on the root, this commit removes the unneeded portions of
the file. Also makes the package.json private to avoid it from being
published from there.

Should anything need be added from a publishing perspective, add it to the
manifests over at scripts/templates.
